pkg/cmd/create_job.go

```diff
@@ -0,0 +1,238 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/d-kuro/kubectl-fuzzy/pkg/fuzzyfinder"
+	"github.com/d-kuro/kubectl-fuzzy/pkg/kubernetes"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/cli-runtime/pkg/resource"
+	batchv1client "k8s.io/client-go/kubernetes/typed/batch/v1"
+	"k8s.io/kubectl/pkg/scheme"
+)
+
+const (
+	exampleCreateJob = `
+	# Selecting a CronJob with the fuzzy finder and create job
+	# Only supported cronjob
+	# If a jobName is omitted, generated from cronJob name
+	kubectl fuzzy create job [jobName] --from=cronjob [flags]
+`
+)
+
+// NewCmdCreateJob provides a cobra command wrapping CreateJobOptions.
+func NewCmdCreateJob(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewCreateJobOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:           "job [NAME] --from=cronjob",
+		Short:         "Selecting a CronJob with the fuzzy finder and create job",
+		Example:       exampleCreateJob,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			if err := o.Run(c.Context()); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	o.configFlags.AddFlags(flags)
+	o.AddFlags(flags)
+
+	return cmd
+}
+
+// CreateJobOptions provides information required to update
+// the current context on a user's KUBECONFIG.
+type CreateJobOptions struct {
+	configFlags       *genericclioptions.ConfigFlags
+	previewPrintFlags *genericclioptions.JSONYamlPrintFlags
+	printFlags        *genericclioptions.PrintFlags
+	genericclioptions.IOStreams
+
+	printObj func(obj runtime.Object) error
+
+	name string
```

Comment:
@d-kuro How do you think about exporting or not the fields of the struct?

I found that it's not exported in sample plugin.
https://github.com/kubernetes/sample-cli-plugin/blob/master/pkg/cmd/ns.go#L49

But it does in kubectl.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/get/get.go#L55-L85

Reply:
I don't think it's necessary to export unless it's referenced from an outside package.
The existing code I wrote is also an unnecessary export. I hope to fix it soon.